src/frontend/src/btc/schedulers/btc-wallet.scheduler.ts

```diff
@@ -46,95 +62,153 @@ export class BtcWalletScheduler implements Scheduler<PostMessageDataRequestBtc>
 		});
 	}
 
-	private async loadBtcTransactionsData({ btcAddress }: { btcAddress: BtcAddress }): Promise<{
-		newTransactions: BitcoinTransaction[];
-		latestBitcoinBlockHeight: number | undefined;
-	}> {
+	private async loadBtcTransactionsData({
+		btcAddress
+	}: {
+		btcAddress: BtcAddress;
+	}): Promise<CertifiedData<BtcTransactionUi>[]> {
 		try {
 			const { txs: fetchedTransactions } = await btcAddressData({ btcAddress });
 
 			const newTransactions = fetchedTransactions.filter(({ hash }) =>
 				isNullish(this.store.transactions[`${hash}`])
 			);
 
-			this.store = {
-				...this.store,
-				transactions: {
-					...this.store.transactions,
-					...newTransactions.reduce(
-						(acc, transaction) => ({
-							...acc,
-							[transaction.hash]: {
-								certified: false,
-								data: transaction
-							}
-						}),
-						{}
-					)
-				}
-			};
-
 			const latestBitcoinBlockHeight = await btcLatestBlockHeight();
 
-			return { newTransactions, latestBitcoinBlockHeight };
+			return newTransactions.map((transaction) => ({
+				data: mapBtcTransaction({ transaction, btcAddress, latestBitcoinBlockHeight }),
+				certified: false
+			}));
 		} catch (error) {
 			// We don't want to disrupt the user experience if we can't fetch the transactions or latest block height.
 			console.error('Error fetching BTC transactions data:', error);
-			// TODO: Return an error instead of an object with empty array.
-			return { newTransactions: [], latestBitcoinBlockHeight: undefined };
+			// TODO: Return an error instead of an empty array.
+			return [];
 		}
 	}
 
-	private async loadBtcBalance({
+	private loadBtcBalance = async ({
 		identity,
-		bitcoinNetwork
-	}: {
-		identity: OptionIdentity;
-		bitcoinNetwork: BitcoinNetwork;
-	}): Promise<CertifiedData<bigint>> {
-		const balance = await getBtcBalance({
-			identity,
-			network: bitcoinNetwork,
-			minConfirmations: BTC_BALANCE_MIN_CONFIRMATIONS
-		});
-		const certifiedBalance = {
-			data: balance,
+		bitcoinNetwork,
+		btcAddress,
+		minterCanisterId,
+		certified = true
+	}: Omit<
+		LoadBtcWalletParams,
+		'shouldFetchTransactions'
+	>): Promise<CertifiedData<bigint> | null> => {
+		// Query BTC balance only if minterCanisterId and BITCOIN_CANISTER_IDS[minterCanisterId] are available
+		if (!certified) {
+			if (nonNullish(minterCanisterId) && BITCOIN_CANISTER_IDS[minterCanisterId]) {
+				return {
+					data: await getBalanceQuery({
+						identity,
+						network: bitcoinNetwork,
+						address: btcAddress,
+						bitcoinCanisterId: BITCOIN_CANISTER_IDS[minterCanisterId]
+					}),
+					certified: false
+				};
+			}
+
+			// corner case in which minter and/or BTC canister ids are not set
```

Comment:
Is that a corner case or should it actually throw an error? i.e. if those are not provided, should the scheduler retries the call later on?

Reply:
I re-worked the code to always return CertifiedData (`bigint | null` in case of balance) as I realized it's not a corner case - we can only query balance for mainnet, therefore it would always end up there for testnet and regtest. Also, added a comment with a small explanation. WDYT?